From 918eaa89f0710af1459de4b40ea884ac1a769710 Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Tue, 18 Jun 2013 12:21:21 +0800
Subject: [PATCH 0646/1302] usb: gadget: f_ecm: fix missing unlock on error in
 ecm_alloc()

Add the missing unlock before return from function ecm_alloc()
in the error handling case.

Introduced by commit da92801c647cdebfd45001fd6aaecb8f0be7f56b.
(usb: gadget: f_ecm: add configfs support)

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: Andrzej Pietrasiewicz <andrzej.p@samsung.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/usb/gadget/f_ecm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/f_ecm.c b/drivers/usb/gadget/f_ecm.c
index fcafe1a..5d3561e 100644
--- a/drivers/usb/gadget/f_ecm.c
+++ b/drivers/usb/gadget/f_ecm.c
@@ -1012,6 +1012,7 @@ struct usb_function *ecm_alloc(struct usb_function_instance *fi)
 					  sizeof(ecm->ethaddr));
 	if (status < 12) {
 		kfree(ecm);
+		mutex_unlock(&opts->lock);
 		return ERR_PTR(-EINVAL);
 	}
 	ecm_string_defs[1].s = ecm->ethaddr;
-- 
1.8.3.2

